dlt/pipeline/pipeline.py

```diff
@@ -1729,3 +1734,11 @@ def _dataset(self, dataset_type: TDatasetType = "dbapi") -> SupportsReadableData
             schema=(self.default_schema if self.default_schema_name else None),
             dataset_type=dataset_type,
         )
+
+    def _ibis(self) -> IbisBackend:
```

Comment:
I don't think our original idea of exposing an ibis dataset via the same methods as the dbapi one makes sense, the interface of ibis is completely different to ours.

Reply:
but you can use overloads on Literal `dataset_type` 
```py
@overload
def _dataset(self, dataset_type: TDatasetType = "dbapi") -> SupportsReadableDataset:
   ...
@overload
def _dataset(self, dataset_type: TDatasetType = "ibis") -> IbisBackend:
   ...
 ```
 to return different types. hmmm but maybe you are right. we should implement readable dataset interface on ibis.
 
 so maybe we add `ibis()` on DBAPI dataset? that will return ibis backend? I do not think adding it on a relation makes sense

Reply:
I like the overload, I did not know this was possible...